client-spark/common/src/main/java/org/apache/spark/shuffle/writer/WriteBufferManager.java

```diff
@@ -551,38 +550,18 @@ public List<AddBlockEvent> buildBlockEvents(List<ShuffleBlockInfo> shuffleBlockI
   @Override
   public long spill(long size, MemoryConsumer trigger) {
     // Only for the MemoryConsumer of this instance, it will flush buffer
-    if (!memorySpillEnabled || trigger != this) {
-      return 0L;
-    }
-
-    List<CompletableFuture<Long>> futures = spillFunc.apply(clear(bufferSpillRatio));
-    CompletableFuture<Void> allOfFutures =
-        CompletableFuture.allOf(futures.toArray(new CompletableFuture[futures.size()]));
-    try {
-      allOfFutures.get(memorySpillTimeoutSec, TimeUnit.SECONDS);
-    } catch (TimeoutException timeoutException) {
-      // A best effort strategy to wait.
-      // If timeout exception occurs, the underlying tasks won't be cancelled.
-      LOG.warn("[taskId: {}] Spill tasks timeout after {} seconds", taskId, memorySpillTimeoutSec);
-    } catch (Exception e) {
-      LOG.warn("[taskId: {}] Failed to spill buffers due to ", taskId, e);
-    } finally {
-      long releasedSize =
-          futures.stream()
-              .filter(x -> x.isDone())
-              .mapToLong(
-                  x -> {
-                    try {
-                      return x.get();
-                    } catch (Exception e) {
-                      return 0;
-                    }
-                  })
-              .sum();
-      LOG.info(
-          "[taskId: {}] Spill triggered by own, released memory size: {}", taskId, releasedSize);
-      return releasedSize;
+    if (memorySpillEnabled && trigger == this) {
+      List<CompletableFuture<Long>> futures = spillFunc.apply(clear(bufferSpillRatio));
+      CompletableFuture<Void> allOfFutures =
+          CompletableFuture.allOf(futures.toArray(new CompletableFuture[futures.size()]));
+      try {
+        // A best effort strategy to wait.
+        allOfFutures.get(memorySpillTimeoutSec, TimeUnit.SECONDS);
+      } catch (Exception e) {
+        // ignore
+      }
     }
+    return 0L;
```

Comment:
From my point, the return value is to tell spark memory manager we have released, that will not duplicate release.

Reply:
I see, but does it really necessary to block here to calculate memory released. since the `free memory` is called in multiply callback thread

Reply:
Sorry, I havn't fully gotten your thought. The returned free memory size is to tell spark the spilled size.

If you don't mind, please provide more details?

Reply:
> `free memory` is called in multiply callback thread
`freeMemory` of `MemoryConsumer` is called in multiply callback thread, but `used` in `MemoryConsumer` is a normal variable,  There is a high probability that concurrent issue happens

Reply:
Emm. I  have to say that the spill only will be triggered by self. 
```

if (!memorySpillEnabled || trigger != this) {
      return 0L;
    }
```